recipes/libcurl/all/conanfile.py

```diff
@@ -76,79 +83,97 @@ def imports(self):
+        # Deprecated options
+        # ===============================
+        self.output.warn("with_openssl, with_winssl, darwin_ssl and with_wolfssl options are deprecated. Use with_ssl option instead.")
+        if tools.is_apple_os(self.settings.os) and self.options.with_ssl == "darwinssl":
+            if self.options.darwin_ssl:
+                self.options.with_ssl == "darwinssl"
+            elif self.options.with_openssl:
+                self.options.with_ssl = "openssl"
+            elif self.options.with_wolfssl:
+                self.options.with_ssl = "wolfssl"
+            else:
+                self.options.with_ssl = False
+        if not tools.is_apple_os(self.settings.os) and self.options.with_ssl == "openssl":
+            if self.settings.os == "Windows" and self.options.with_winssl:
+                self.options.with_ssl = "winssl"
+            elif self.options.with_openssl:
+                self.options.with_ssl = "openssl"
+            elif self.options.with_wolfssl:
+                self.options.with_ssl = "wolfssl"
+            else:
+                self.options.with_ssl = False
+        del self.options.with_openssl
+        del self.options.with_winssl
+        del self.options.darwin_ssl
+        del self.options.with_wolfssl
+        # ===============================
+
+        if self.options.with_ssl == "winssl" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("winssl only suppported on Windows.")
+        if self.options.with_ssl == "darwinssl" and not tools.is_apple_os(self.settings.os):
+            raise ConanInvalidConfiguration("darwinssl only suppported on Apple like OS (Macos, iOS, watchOS or tvOS).")
+
+        # enforce shared linking due to openssl dependency
+        if self.options.with_ssl == "openssl":
+            self.options["openssl"].shared = self.options.shared
+        elif self.options.with_ssl == "wolfssl":
+            self.options["wolfssl"].shared = self.options.shared
+        if self.options.with_libssh2:
+            self.options["libssh2"].shared = self.options.shared
 
-        if self.options.with_openssl and self.options.with_wolfssl:
-            raise ConanInvalidConfiguration("Specify either with_openssl or with_wolfssl")
+        # These options are not used in CMake build
+        if self.settings.compiler == "Visual Studio" or self._is_win_x_android:
+            del self.options.with_libidn
+            del self.options.with_librtmp
+            del self.options.with_libmetalink
+            del self.options.with_libpsl
+            del self.options.with_nghttp2
+            del self.options.with_brotli
 
-        if self.options.with_openssl:
-            # enforce shared linking due to openssl dependency
-            if not tools.is_apple_os(self.settings.os) or not self.options.darwin_ssl:
-                self.options["openssl"].shared = self.options.shared
+    def requirements(self):
+        self.requires("zlib/1.2.11")
```

Comment:
please see 
https://stackoverflow.com/questions/23772816/when-do-i-need-zlib-in-openssl/23829863

I am not sure that compression should be the default